### PR TITLE
[KOGITO-3553] - Ignore Self signed certificates from maven while bui…

### PR DESCRIPTION
**Pull request by Kaitou786**
…lding images

The PR aims to provide a mechanism to relax the SSL for user-generated self-signed certificates, Currently, our image build fails when a user has a maven mirror with self-signed certificates, after the PR setting `MAVEN_IGNORE_SELF_SIGNED_CERTIFICATE` to "true" relax the constraint making our image to build successfully with self-signed maven mirror's certificate. For more info please [see](https://stackoverflow.com/questions/21252800/how-to-tell-maven-to-disregard-ssl-errors-and-trusting-all-certs)
See: https://issues.redhat.com/browse/KOGITO-3553

Signed-off-by: Tarun Khandelwal <tarkhand@redhat.com>

Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors guide](README.md#contributing-to-kogito-images-repository)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a testcase that verifies it
- [ ] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

**Review by vaibhavjainwiz — APPROVED** [review]
LGTM

File: modules/kogito-maven/3.6.x/added/configure-maven.sh [hunk 1]
```diff
@@ -19,12 +19,14 @@ function prepareEnv() {
     unset MAVEN_REPO_SNAPSHOTS_CHECKSUM_POLICY
     unset MAVEN_REPO_URL
     unset MAVEN_REPOS
+    unset IGNORE_SELF_SIGNED_CERTIFICATE
```
Comment (radtriste) [inline modules/kogito-maven/3.6.x/added/configure-maven.sh]: thinking that we should maybe prefix it with `MAVEN` as this is maven particularity.
wdyt ?

File: modules/kogito-maven/3.6.x/module.yaml [hunk 2]
```diff
@@ -62,7 +62,9 @@ envs:
   - name: "MAVEN_REPOS"
     description: "Used to define multiple repositories, this env defines a prefix that will be used to create different repositories."
     example: "CENTRAL,INTERNAL"
-
+  - name: "IGNORE_SELF_SIGNED_CERTIFICATE"
+    description: "When set, use of relaxed SSL check for user generated certificates. Default value is false"
+    value: "true"
```
Comment (radtriste) [inline modules/kogito-maven/3.6.x/module.yaml]: `Default value is false` and then `value: "true"` :)
just put an example and not a value

Comment (Kaitou786) [reply modules/kogito-maven/3.6.x/module.yaml]: Ah yes, :p, missed this, will change it

File: modules/kogito-maven/3.6.x/added/configure-maven.sh (hunk 1 above)
Comment (Kaitou786) [reply modules/kogito-maven/3.6.x/added/configure-maven.sh]: It is part of, maven module but I think as other vars have that as a prefix I'll add it to this one :)

File: modules/kogito-maven/3.6.x/added/configure-maven.sh [hunk 3]
```diff
@@ -19,12 +19,14 @@ function prepareEnv() {
     unset MAVEN_REPO_SNAPSHOTS_CHECKSUM_POLICY
     unset MAVEN_REPO_URL
     unset MAVEN_REPOS
+    unset MAVEN_IGNORE_SELF_SIGNED_CERTIFICATE
```
Comment (spolti) [inline modules/kogito-maven/3.6.x/added/configure-maven.sh]: please keep in alphabetical order.

File: modules/kogito-maven/3.6.x/added/configure-maven.sh [hunk 4]
```diff
@@ -19,12 +19,14 @@ function prepareEnv() {
...
 }
 
 function configure() {
     configure_proxy
     configure_mirrors
     configure_maven_download_output
+    configure_self_signed_certificates
```
Comment (radtriste) [inline modules/kogito-maven/3.6.x/added/configure-maven.sh]:
```suggestion
    configure_maven_self_signed_certificates
```

File: modules/kogito-maven/3.6.x/added/configure-maven.sh [hunk 5]
```diff
@@ -92,6 +94,12 @@ function configure_maven_download_output() {
     fi
 }
 
+function configure_self_signed_certificates() {
```
Comment (radtriste) [inline modules/kogito-maven/3.6.x/added/configure-maven.sh]:
```suggestion
function configure_maven_self_signed_certificates() {
```

File: modules/kogito-maven/3.6.x/added/configure-maven.sh (hunk 4 above)
Comment (spolti) [inline modules/kogito-maven/3.6.x/added/configure-maven.sh]: I would change this function to ignore_self

**Review by spolti — CHANGES_REQUESTED** [review]

**Review by spolti — APPROVED** [review]

**Comment by radtriste** [comment]
For the pipeline, I think we would need a new possibility in `scripts/update-tests.py` to disable certificate verification and that would add `| MAVEN_IGNORE_SELF_SIGNED_CERTIFICATE | true |` into the env of feature tests (see how it is done with MAVEN_REPO_URL for example).
And then in pipeline, before launching the tests, if `env.MAVEN_MIRROR_URL` is defined, then we should update the tests with the certificate disabled.

**Comment by Kaitou786** [comment]
> For the pipeline, I think we would need a new possibility in `scripts/update-tests.py` to disable certificate verification and that would add `| MAVEN_IGNORE_SELF_SIGNED_CERTIFICATE | true |` into the env of feature tests (see how it is done with MAVEN_REPO_URL for example).

Okay, will add the function, this should be part of this PR or another small JIRA?

**Comment by radtriste** [comment]
I guess you can implement as part of this PR as it is related

File: Jenkinsfile [hunk 6]
```diff
@@ -37,6 +37,11 @@ pipeline{
                     }
 
                     githubscm.checkoutIfExists('kogito-images', changeAuthor, changeBranch, 'kiegroup', changeTarget, true)
+
+                    //Ignore self-signed certificates if MAVEN_MIRROR_URL is defined
+                    if(env.MAVEN_MIRROR_URL != ""){
```
Comment (radtriste) [inline Jenkinsfile]: please use single-quote

File: Jenkinsfile [hunk 7]
```diff
@@ -37,6 +37,11 @@ pipeline{
...
+                        sh "python3 scripts/update-tests.py --ignore-self-signed-cert=True"
```
Comment (radtriste) [inline Jenkinsfile]: you should not need to have `=True`. if option is there, then it is true

File: scripts/common.py [hunk 8]
```diff
@@ -222,7 +222,15 @@ def update_maven_repo_in_behave_tests(repo_url, replaceJbossRepository):
         envVarKey = "JBOSS_MAVEN_REPO_URL"
     replacement = "| variable | value |\n      | {} | {} |\n      | MAVEN_DOWNLOAD_OUTPUT | true |".format(envVarKey, repo_url)
     update_in_behave_tests(pattern, replacement)
-
+def ignore_maven_self_signed_certificate_in_behave_tests():
+    """
+    Sets the enviornment variable to ignore the self-signed certificates in maven
```
Comment (radtriste) [inline scripts/common.py]:
```suggestion
    Sets the environment variable to ignore the self-signed certificates in maven
```

File: scripts/update-tests.py [hunk 9]
```diff
@@ -17,6 +17,7 @@
     parser.add_argument('--examples-uri', dest='examples_uri', help='To update the examples uri for testing')
     parser.add_argument('--examples-ref', dest='examples_ref', help='To update the examples ref for testing')
     parser.add_argument('--artifacts-version', dest='artifacts_version', help='To update the artifacts version for testing')
+    parser.add_argument('--ignore-self-signed-cert', dest='ignore_self_signed_cert', help='If set to true will relax the SSL for user-generated self-signed certificates', default=False)
```
Comment (radtriste) [inline scripts/update-tests.py]: see `--replace-jboss-repo` option to have similar behavior

File: Jenkinsfile [hunk 10]
```diff
@@ -37,6 +37,11 @@ pipeline{
                     }
 
                     githubscm.checkoutIfExists('kogito-images', changeAuthor, changeBranch, 'kiegroup', changeTarget, true)
+
+                    //Ignore self-signed certificates if MAVEN_MIRROR_URL is defined
+                    if(env.MAVEN_MIRROR_URL != ''){
+                        sh "python3 scripts/update-tests.py --ignore-self-signed-cert"
```
Comment (radtriste) [inline Jenkinsfile]: as long as you are making usage of gstring (which means include a variable of a string with `${VAR}`), you should use single-quote ;)
```suggestion
                        sh 'python3 scripts/update-tests.py --ignore-self-signed-cert'
```

File: Jenkinsfile.deploy [hunk 11]
```diff
@@ -206,13 +206,17 @@ pipeline {
...
+                    if(params.EXAMPLES_REF != ''){
                         updateTestsCommand += " --examples-ref ${params.EXAMPLES_REF}"
                     }
                     if(params.EXAMPLES_URI){
                         updateTestsCommand += " --examples-uri ${params.EXAMPLES_URI}"
                     }
 
+                    //Ignore self-signed certificates if MAVEN_MIRROR_URL is defined
+                    if(env.MAVEN_MIRROR_URL != ''){
+                        updateTestsCommand += " --ignore-self-signed-cert"
```
Comment (radtriste) [inline Jenkinsfile.deploy]:
```suggestion
                        updateTestsCommand += ' --ignore-self-signed-cert'
```

**Review by radtriste — CHANGES_REQUESTED** [review]
gonna test this tomorrow

File: Jenkinsfile (hunk 10 above)
Comment (Kaitou786) [reply Jenkinsfile]: Ah okay, thanks, will keep in mind from next time :)

**Review by radtriste — APPROVED** [review]

**Comment by radtriste** [comment]
Tested ok with a specific MAVEN_MIRROR_URL with self signed certificate

**Pull request merged**